strawberryfields/decompositions.py

```diff
@@ -476,6 +476,190 @@ def rectangular_symmetric(V, tol=1e-11):
     return new_tlist, new_diags, None
 
 
+def BS_mat(theta):
+    r"""Beam splitter matrix as implemented in hardware
+    """
+    mat = np.array([[np.cos(theta), 1j*np.sin(theta)], [1j*np.sin(theta), np.cos(theta)]])
+    return mat
+
+def R(phi):
+    r"""Phase as implemented in hardware
+    """
+    mat = np.array([[np.exp(1j*phi), 0], [0, 1]])
+    return mat
+
+def MZ(m, n, phi_int, phi_ext, nmax):
+    r"""One unit as implemented on the chip. 
+    An external phase followed by a beamsplitter followed by 
+    the internal phase followed by another beamsplitter
+    """
+    theta = np.pi/4
+    BS1 = BS_mat(theta)
+    BS2 = BS_mat(theta)
+    T = BS1 @ R(phi_int) @ BS2 @ R(phi_ext)
+    
+    mat = np.identity(nmax, dtype=np.complex128)
+    mat[m, m] = T[0][0]
+    mat[m, n] = T[0][1]
+    mat[n, m] = T[1][0]
+    mat[n, n] = T[1][1]
+    return mat
```

Comment:
This seems like a duplicate of the `mach_zehnder` function above (line 383); if this (plus `BS` and `R`) replace that function, perhaps best to delete the old `mach_zehnder` function and rename this one to `mach_zehnder` (and make sure that the signature remains the same!)